tools/memory-model-test/README.md

```diff
@@ -0,0 +1,201 @@
+# WiredTiger Memory Model Test
+
+The WiredTiger Memory Model Test performs a series of operations that, depending on the processor,
+may generate out-of-order memory accesses that are visible to the code.
+
+It is not a 'pass or fail' type of test. It is a technology demo and testbed for what can occur with
+out-of-order memory accesses, and clearly shows in concrete form the differences in behaviour
+between the ARM64 and x86_64 memory models.
+
+This test was inspired, in part, by the work at https://preshing.com/20120515/memory-reordering-caught-in-the-act/, 
+which is worth reading to understand these tests.
+
+## The Tests
+
+Memory Model Test runs a series of tests. Each test has two threads which are accessing shared variables. 
+
+All shared variables are set to 0 before each test.
+
+There are two broad groups of tests:
+- Group 1: 
+  - Each thread writes 1 to a shared variable (`x` for thread 1, and `y` for thead 2),
+    and then reads from the other shared variable (`y` for thread 1, and `x` for thead 2)
+  - Variants of the test are run with no barriers, with one barrier or atomic, or with two barriers or atomics
+  - If out-of-order memory accesses occur, then the results of the reads of `x` and `y` can lead to both being 0.
+- Group 2:
+  - Thread 1 writes 2 to `x` and then 3 to `y`, while thread 2 reads from `y` and then reads from `x`
+  - Variants of the test are run with no barriers, with one barrier or atomic, or with two barriers or atomics
+  - If out-of-order memory accesses occur, then the results of the reads of `y` being 3, and of `x` being 0.
+
+
+## Building and running the tests
+
+### Requirements
+* ARM64 or x86_64 based CPU
+* C++ compiler that supports C++ 20
+* (optional) CMake and ninja
+
+### Build
+
+There are two options:
+
+* Use the CMakeLists.txt file with CMake:
+  ```
+  md build
```

Comment:
I think `md` is a Jeremy's shell specific command ;-)

Reply:
Well spotted - now fixed.